### PR TITLE
Get data

### PR DESCRIPTION
**Pull request by Dev-AnChi**

### CODE REVIEW

**Comment by Dev-AnChi** [comment]
merge

File: app/controllers/dimension_histories_controller.rb [hunk 1]
```diff
@@ -1,17 +1,77 @@
...
-        # @dimension_history = DimensionHistory.find(params[:id])
-    end
+  def load_dimensions
+    # Lấy tất cả kết quả từ bảng DimensionHistory để hiển thị
+    @dimension_histories = DimensionHistory.all
+  end
+
+  def generate_random_json(height, width, length, weight)
+    # Tạo dữ liệu mẫu ngẫu nhiên từ các giá trị nhập vào
+    data = {
```
Comment (quanld1502) [inline app/controllers/dimension_histories_controller.rb:58]: { datetime: Time.now.strftime("%Y-%m-%d %H:%M:%S"), ... }

File: app/controllers/dimension_histories_controller.rb [hunk 2]
```diff
@@ -1,17 +1,77 @@
...
+    # Hiển thị JSON đẹp mắt
+    @api_result = api_result.present? ? JSON.pretty_generate(api_result) : nil
+    respond_to do |format|
+      format.json { render json: @api_result }  # Sửa lại để trả về @api_result
+      format.js   # Phản hồi định dạng JavaScript
     end
+  rescue StandardError => e
+    # Ghi log cho lỗi
+    puts "Lỗi : #{e.message}"
+    flash[:alert] = 'Có lỗi.'
```
Comment (quanld1502) [inline app/controllers/dimension_histories_controller.rb:45]: :alert & "alert"

File: app/controllers/dimension_histories_controller.rb [hunk 3]
```diff
@@ -1,17 +1,77 @@
...
+  def get_dimensions
+    # Tạo dữ liệu ngẫu nhiên từ các giá trị nhập vào
+    api_result = generate_random_json(params[:height], params[:width], params[:length], params[:weight])
+
+    # Kiểm tra xem có kết quả hay không
+    if api_result.present?
+      # Lưu kết quả vào bảng DimensionHistory
+      @dimension_history = DimensionHistory.create(
+        url: params[:url],
+        datetime: api_result["datetime"],
```
Comment (quanld1502) [inline app/controllers/dimension_histories_controller.rb:18]: api_result[:datetime]

File: app/controllers/dimension_histories_controller.rb [hunk 4]
```diff
@@ -1,17 +1,77 @@
+require 'json'
+
 class DimensionHistoriesController < ApplicationController
-    before_action :set_dimension_history, only: [:show, :index]
+  before_action :load_dimensions, only: [:index, :get_dimensions]
```
Comment (quanld1502) [inline app/controllers/dimension_histories_controller.rb:4]: remove `get_demensions`

File: app/views/dimension_histories/index.html.erb [hunk 5]
```diff
@@ -1,67 +1,69 @@
-<!-- app/views/dimension_histories/index.html.erb -->
+<div class="container mt-4">
+  <h1>Form</h1>
 
-<h1 class="mt-4">Form</h1>
+  <%= form_tag({ controller: 'dimension_histories', action: 'get_dimensions' }, method: 'get', class: 'mb-4', remote: true, id: 'dimension-history-form') do %>
```
Comment (quanld1502) [inline app/views/dimension_histories/index.html.erb:4]: Please use `form_with`

**Pull request merged**